File: src/main/java/de/steklopod/sec09/Lec01Buffer.java
package de.steklopod.sec09;

import de.steklopod.courseutil.Util;
import reactor.core.publisher.Flux;

import java.time.Duration;

public class Lec01Buffer {

    public static void main(String[] args) {

        eventStream()
                .bufferTimeout(5, Duration.ofSeconds(2))
                .subscribe(Util.subscriber());

        Util.sleepSeconds(60);

    }


    private static Flux<String> eventStream(){
        return Flux.interval(Duration.ofMillis(800))
                    .map(i -> "event"+i);
    }


}

File: src/main/kotlin/de/steklopod/sec09/Lec01Buffer.kt
package de.steklopod.sec09

import de.steklopod.utils.Util.sleepSeconds
import de.steklopod.utils.Util.subscriber
import reactor.core.publisher.Flux
import java.time.Duration

object Lec01Buffer {
    @JvmStatic
    fun main(args: Array<String>) {
        eventStream()
            .bufferTimeout(5, Duration.ofSeconds(2))
            .subscribe(subscriber())

        sleepSeconds(60)
    }

    private fun eventStream(): Flux<String> = Flux.interval(Duration.ofMillis(800))
        .map { i: Long -> "event$i" }
}
